tests/unit/cluster/import.tcl

```diff
@@ -0,0 +1,1402 @@
+    foreach migration $migrations {
+        set slot_ranges [dict get $migration slot_ranges]
+        if {[slot_ranges_contains_slot $slot_ranges $slot]} {
+            return [dict get $migration link_name]
+        }
+    }
+    return ""
+}
+
+proc get_migration_by_linkname {node_idx linkname} {
+    set migrations [R $node_idx CLUSTER MIGRATIONS]
+    foreach migration $migrations {
+        if {[dict get $migration link_name] eq $linkname} {
+            return $migration
+        }
+    }
+    return ""
+}
+
+proc wait_for_migration_field {node_idx linkname field value} {
+    wait_for_condition 100 100 {
+        [get_migration_by_linkname $node_idx $linkname] ne "" && [dict get [get_migration_by_linkname $node_idx $linkname] $field] eq $value
+    } else {
+        set curr_state [get_migration_by_linkname $node_idx $linkname]
+        fail "Migration $linkname on node $node_idx did not have $field == $value (currently $curr_state) within 10000 ms"
+    }
+}
+
+proc wait_for_countkeysinslot {node_idx slot value} {
+    wait_for_condition 100 100 {
+        [R $node_idx CLUSTER COUNTKEYSINSLOT $slot] eq "$value"
+    } else {
+        set curr_count [R $node_idx CLUSTER COUNTKEYSINSLOT $slot]
+        fail "Node $node_idx did not have $value keys in slot $slot within 10000 ms (current $curr_count)"
+    }
+}
+
+proc wait_for_migration {node_idx slot} {
+    set target_id [R $node_idx CLUSTER MYID]
+    wait_for_condition 100 100 {
+        [is_slot_migrated $node_idx $slot]
+    } else {
+        set nodes [get_cluster_nodes $node_idx]
+        fail "Cluster node $target_id did not get slot $slot within 10000 ms (current $nodes)"
+    }
+    wait_for_cluster_propagation
+}
+
+proc get_cluster_total_syncs_count {} {
+    set total 0
+    foreach node {0 1 2 3 4 5} {
+        set total [expr [status [Rn $node] sync_full] + $total]
+        set total [expr [status [Rn $node] sync_partial_ok] + $total]
+        set total [expr [status [Rn $node] sync_partial_err] + $total]
+    }
+    return $total
+}
+
+# Helper to wrap a test, and assert it doesn't cause a resync
+proc assert_does_not_resync {body} {
+    set prev_syncs [get_cluster_total_syncs_count]
+    uplevel 1 $body
+    assert_equal $prev_syncs [get_cluster_total_syncs_count]
+}
+
+proc assert_causes_syncslots_fail {node_idx args} {
+    set client [valkey_client_by_addr [srv -$node_idx host] [srv -$node_idx port]]
+    assert_match "*CLUSTER SYNCSLOTS FAIL*" [$client {*}$args]
+    $client deferred 1
+    catch {
+        $client read
+    } result
+    $client deferred 0
+    $client close
+    assert_match "*I/O error reading reply*" $result
+}
+
+proc assert_causes_conn_drop {node_idx args} {
+    set client [valkey_client_by_addr [srv -$node_idx host] [srv -$node_idx port]]
+    catch {
+        $client {*}$args
+    } result
+    $client close
+    assert_match "*I/O error reading reply*" $result
+}
+
+test "General command interface" {
```

Comment:
This entire test should be indented since it's under the start cluster. Some cluster tests aren't indented since it messed up git history and caused backporting issues, but new tests can be indented normally.